alacritty/src/window_context.rs

```diff
@@ -92,9 +92,10 @@ impl WindowContext {
         // The PTY forks a process to run the shell on the slave side of the
         // pseudoterminal. A file descriptor for the master side is retained for
         // reading/writing to the shell.
-        let pty_config = options
-            .map(|o| Cow::Owned(o.into()))
-            .unwrap_or(Cow::Borrowed(&config.terminal_config.pty_config));
+        let mut pty_config = Cow::Borrowed(&config.terminal_config.pty_config);
+        if let Some(o) = options {
+            pty_config.to_mut().working_directory = o.working_directory;
+        }
```

Comment:
This behavior is incorrect, it will not work with the new multiwindow work.

Any patch fixing this issue should be **outside** of this function before it is passed inside. That's why I suggested to base it on the existing PR changing the working directory behavior.